metricbeat/module/vsphere/virtualmachine/virtualmachine.go

```diff
@@ -132,28 +132,46 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
 	}
 
 	for _, vm := range vmt {
-		freeMemory := (int64(vm.Summary.Config.MemorySizeMB) * 1024 * 1024) - (int64(vm.Summary.QuickStats.GuestMemoryUsage) * 1024 * 1024)
+		totalMemory := int64(vm.Summary.Config.MemorySizeMB) * 1024 * 1024
+		usedMemory := int64(vm.Summary.QuickStats.GuestMemoryUsage) * 1024 * 1024
+		freeMemory := totalMemory - usedMemory
+		if freeMemory < 0 {
```

Comment:
I wonder that if this situation and in one next for `freeCPU` it's better to not report the metric at all. Because 0 here will be a reported value, a user will expect that it doesn't have free memory, maybe the user will configure some alerts on this. But it might just be a temporal network error, so Beats report will be wrong and misleading.

I don't know, WDYT? :smile:

Reply:
I added these safe-checks to avoid having negative values if for some reason the value reported for the used resource is greater than the total one, something that shouldn't happen, and in principle doesn't happen for memory. For CPU I have seen values of entitled MHz slightly higher than the configured ones, so it might happen, and in these cases I think it would make more sense to report 0 than a small negative value.

All the metrics are reported in the same request, so regarding network errors it'd be an all or nothing.

Something that could happen is that there is a 0 configured value for CPU or memory, in these case total and free would be always 0 and in this case it can make sense to don't report anything. I will add this case, but not sure if can happen in a real scenario.

Reply:
Comment added too to explain why this is being checked.

Reply:
Yeah, sounds good too :)